Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_installation.py

```diff
@@ -0,0 +1,204 @@
+
+        #TODO replace with release that has the Editor Commands changes
+        manifest_entry = "0.14.5-preview"
+        if DEBUG == 1:
+            manifest_entry = "file:C:\\Users\\Sean Dillon\\MeshSync\\MeshSync~\\Packages\\com.unity.meshsync"
+
+        msb_add_meshsync_to_unity_manifest(manifest_path, lock_path, manifest_entry)
+
+def msb_try_setup_scene_server(context):
+
+    # Try get valid path
+    path = msb_try_get_valid_project_path(context)
+    if path == 'INVALID_PATH':
+        return 'INVALID_PATH'
+
+    # Write back the valid path, in case it comes from a listening server
+    context.scene.meshsync_unity_project_path = path
+
+    # Try install to unity project
+    msb_try_install_meshsync_to_unity_project(path)
+
+    # Try starting unity if not already started
+    start_status = msb_try_start_unity_project(path)
+
+    #If the project was launched now, wait until the Editor server is available
+    if start_status == 'STARTED':
+        while(msb_context.is_editor_server_available is False):
+            time.sleep(0.1)
+    elif start_status == 'EDITOR_NOT_EXISTS':
+        return 'EDITOR_NOT_EXISTS'
+
+    # Send a command to add a scene server (if it doesn't exist already)
+    msb_context.sendEditorCommand(EDITOR_COMMAND_ADD_SERVER)
+    reply = msb_context.editor_command_reply
+
+    if not reply == 'ok':
+        return 'SERVER_NOT_ADDED'
+
+    return 'SUCCESS'
+
+def msb_get_editor_version(directory):
+    project_version_path = directory + "/ProjectSettings/ProjectVersion.txt"
+    with open(project_version_path, "r+") as file:
+        first_line = file.readline()
+        version = first_line[len("m_EditorVersion: "):]
+        version = version[:-1]
+        return version
+
+def msb_launch_project(editor_path, project_path):
+    os = platform.system()
+    if os == 'Windows':
+        path = editor_path + " -projectPath \"" + project_path + "\""
+    elif os == 'Darwin' or os == 'Linux':
+        path = editor_path + " -projectPath " + project_path
+
+    return subprocess.Popen(path)
+
+def msb_try_start_unity_project (directory):
+        
+    global unity_process
+
+    #Check if we have launched the project as a subprocess
+    if unity_process is not None and unity_process.poll() is None:
+        return 'ALREADY_STARTED'
+
+    #Check if there is an editor server listening from the target project
+    if msb_context.is_editor_server_available:
+        msb_context.sendEditorCommand(EDITOR_COMMAND_GET_PROJECT_PATH)
+        reply_path = msb_context.editor_command_reply;
+        if reply_path == directory:
+            return 'ALREADY_STARTED'
+
+    editor_version = msb_get_editor_version(directory)
+    editor_path = msb_get_editor_path(editor_version)
+
+    if not path.exists(editor_path):
+        return 'EDITOR_NOT_EXISTS'
+
+    #Launch the editor with the project
+    unity_process = msb_launch_project(editor_path, directory)
```

Comment:
Could this process launch fail? Is there a result or bool or anything you can check here to make sure the process successfully started?

Reply:
> Could this process launch fail?
We have checked that the path to the Editor .exe exists before getting here. How would the launch fail? More importantly, how are we supposed to handle it?

Reply:
I'm not sure, I just mean if the application failed to start up for some reason (maybe it needs a permission or something - just thinking of possibilities where the process can't start the exe). We should display something to the user saying "Failed to launch editor" so they know they manually need to go and open Unity

Reply:
It may not be possible, I was just guessing that `subprocess.Popen()` might return a success code or some indication that the process started/failed

Reply:
> might return a success code or some indication that the process started/failed

Turns out it throws an exception if something goes wrong. I tested by modifying the path to something that does not exists before calling it. The idea to notify the user sounds good to me, better than getting the error stack. I have now implemented it